internal/runner/rspec.go

```diff
@@ -63,64 +67,72 @@ func (r Rspec) GetFiles() ([]string, error) {
 	return files, nil
 }
 
-func (r Rspec) RetryCommand() (*exec.Cmd, error) {
-	words := []string{}
-
-	if r.RetryTestCommand == "" {
-		// use test command to build retry command if retry command is not provided
-		// remove all occurrences of "{{testExamples}}" from the test command and append "--only-failures"
-		splits, err := shellquote.Split(r.TestCommand)
-		if err != nil {
-			return nil, err
-		}
-		splits = slices.DeleteFunc(splits, func(n string) bool {
-			return n == "{{testExamples}}"
-		})
-		splits = slices.Insert(splits, len(splits), "--only-failures")
-		words = append(words, splits...)
+// Run executes the test command with the given test cases.
+// If retry is true, it will run the command using the retry test command,
+// otherwise it will use the test command.
+//
+// Error is returned if the command fails to run, exits prematurely, or if the
+// output cannot be parsed.
+//
+// Test failure is not considered an error, and is instead returned as a RunResult.
+func (r Rspec) Run(testCases []string, retry bool) (RunResult, error) {
+	var command string
+
+	if retry {
+		command = r.RetryTestCommand
```

Comment:
(Very low-importance suggestion)

```suggestion
	command := r.TestCommand
	if retry {
		command = r.RetryTestCommand
	}
```

(no `else` needed!)